tls/extensions/s2n_client_server_name.c

```diff
@@ -22,67 +22,85 @@
 
 #include "utils/s2n_safety.h"
 
-int s2n_extensions_client_server_name_send(struct s2n_connection *conn, struct s2n_stuffer *out)
+#define S2N_NAME_TYPE_HOST_NAME 0
+
+static bool s2n_client_server_name_should_send(struct s2n_connection *conn);
+static int s2n_client_server_name_send(struct s2n_connection *conn, struct s2n_stuffer *out);
+static int s2n_client_server_name_recv(struct s2n_connection *conn, struct s2n_stuffer *extension);
+
+const s2n_extension_type s2n_client_server_name_extension = {
+    .iana_value = TLS_EXTENSION_SERVER_NAME,
+    .is_response = false,
+    .send = s2n_client_server_name_send,
+    .recv = s2n_client_server_name_recv,
+    .should_send = s2n_client_server_name_should_send,
+    .if_missing = s2n_extension_noop_if_missing,
+};
+
+static bool s2n_client_server_name_should_send(struct s2n_connection *conn)
 {
-    uint16_t server_name_len = strlen(conn->server_name);
-
-    /* Write the server name */
-    GUARD(s2n_stuffer_write_uint16(out, TLS_EXTENSION_SERVER_NAME));
-    GUARD(s2n_stuffer_write_uint16(out, server_name_len + 5));
+    return strlen(conn->server_name) > 0;
```

Comment:
I'm sure `conn` was notnull checked before we got to this function, but recent reviews have wanted notnull() checks on all pointer arguments so we can maintain consistency.

Reply:
I can put it in there for consistency, but it's pretty unnecessary in extensions. s2n_extension_send/recv does the standard null checks before calling should_send/send/recv.